### ISSUE TITLE
Oasis hangs with spinners when launched with fresh, locked Metamask

### ISSUE THREAD

**Issue by livnev** [OasisDEX/oasis#326]
Causes quite a hostile UX. Steps to reproduce:

1. Close Oasis
2. Disable and enable Metamask from the Chrome Extensions menu, but do not unlock Metamask
3. Navigate to oasisdex.com, you will see a page asking you to unlock Metamask
4. Do as it says and unlock Metamask, but do not refresh the page
5. The page will try to load, but will eventually get stuck with spinners (see screenshot) while hogging a full CPU core. Eventually Chrome will offer to kill the tab.
6. After killing the tab and returning to oasisdex.com, everything will work normally.

![oasishang](https://user-images.githubusercontent.com/3964494/29841742-84e129e2-8cfe-11e7-8048-813a1add8537.png)

**Comment by peculiarity** [comment]
I've tried to reproduce it and it's reproducible only on **mainnet**

**Comment by dalaei** [comment]
I am still having this issue and also unable to wrap/unwrap withdraw or transfer. Ive tried the suggestion in the OP but still not working.

**Comment by pimato** [comment]
@dalaei where do you stugged exactly ? 

Is oasis not loading up ?
What kind of problems do you have regarding wrap/unwrap ?

**Comment by pimato** [comment]
@livnev do you still experience the same issues ?

**Comment by livnev** [comment]
@pimato Yes, I am still able to reproduce using the same steps. However, at the moment instead of getting the spinners, it is just stuck on the "Metamask Account Locked" page and hogs a full CPU thread.

**Comment by peculiarity** [comment]
I had couple of issues with parity as well. Sometimes on first load the whole page gets stuck with the spinners and the CPU goes over 100%. Needs investigation in depth

**Comment by peculiarity** [comment]
@livnev  Can you confirm if this still persists? And if so please provide versions of the browser and the metamask-plugin.

**Comment by livnev** [comment]
@peculiarity Yep, reproducing the same issue.

- Chromium Version 61.0.3163.100 (Developer Build) built on Debian buster/sid, running on Debian buster/sid (64-bit)
- Metamask 3.12.0

**Comment by peculiarity** [comment]
Well I've tried it and didn't have problem though I had a case where the CPU increased and there were spinners but I just had to wait little bit longer. Will try to reproduce it then.

How long have you waited btw ?

**Comment by pimato** [comment]
@livnev I have the exact same experience with:

- Chrome Version 62.0.3202.94 (Offizieller Build) (64-Bit)
- Metamask 3.12.0

We are currently transition from meteor to react, maybe that will fix the bug. Otherwise we are going to dig deeper into it.
